src/pages/settings/Wallet/VerifyAccountPage.tsx

```diff
@@ -0,0 +1,90 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback, useEffect, useRef} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import type {AnimatedTextInputRef} from '@components/RNTextInput';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import ValidateCodeForm from '@components/ValidateCodeActionModal/ValidateCodeForm';
+import useLocalize from '@hooks/useLocalize';
+import useSafePaddingBottomStyle from '@hooks/useSafePaddingBottomStyle';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import * as User from '@userActions/User';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+type VerifyAccountPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.PROFILE.NEW_CONTACT_METHOD>;
+
+function VerifyAccountPage({route}: VerifyAccountPageProps) {
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
+    const contactMethod = account?.primaryLogin ?? '';
+    const themeStyles = useThemeStyles();
+    const {translate} = useLocalize();
+    const safePaddingBottomStyle = useSafePaddingBottomStyle();
+    const loginInputRef = useRef<AnimatedTextInputRef>(null);
+    const firstRenderRef = useRef(true);
+    const loginData = loginList?.[contactMethod];
+    const styles = useThemeStyles();
+    const validateLoginError = ErrorUtils.getEarliestErrorField(loginData, 'validateLogin');
+    const [isUserValidated] = useOnyx(ONYXKEYS.USER, {selector: (user) => !!user?.validated});
+
+    const [validateCodeAction] = useOnyx(ONYXKEYS.VALIDATE_ACTION_CODE);
+
+    const navigateBackTo = route?.params?.backTo ?? ROUTES.SETTINGS_WALLET;
+
+    useEffect(() => {
+        if (firstRenderRef.current) {
```

Comment:
```suggestion
```

`firstRenderRef` should be unnecessary here. This effect would only run once after the first render so there's no need to use a ref to check.